pkg/client/observatorium/client_test.go

```diff
@@ -0,0 +1,358 @@
+			fields: fields{
+				p: observatoriumRoundTripper{
+					config: config,
+					wrapped: observatoriumRoundTripper{
+						config:  config,
+						wrapped: pAPI.DefaultRoundTripper,
+					},
+				},
+			},
+			args: args{
+				request: req,
+			},
+			wantErr: false,
+		},
+		{
+			name: "should return no error with roundTripper's AuthType set to AuthTypeDex and sample valid request and non-empty AuthToken",
+			fields: fields{
+				p: observatoriumRoundTripper{
+					config: config,
+					wrapped: observatoriumRoundTripper{
+						config:  config,
+						wrapped: pAPI.DefaultRoundTripper,
+					},
+				},
+			},
+			args: args{
+				request: req,
+			},
+			modifyFn: func(config *observatoriumRoundTripper) {
+				config.config.AuthType = AuthTypeDex
+				config.config.AuthToken = testValue
+			},
+			wantErr: false,
+		},
+		{
+			name: "should return no error with roundTripper's AuthType set to AuthTypeDex and sample valid request and non-empty Cookie",
+			fields: fields{
+				p: observatoriumRoundTripper{
+					config: config,
+					wrapped: observatoriumRoundTripper{
+						config:  config,
+						wrapped: pAPI.DefaultRoundTripper,
+					},
+				},
+			},
+			args: args{
+				request: req,
+			},
+			modifyFn: func(config *observatoriumRoundTripper) {
+				config.config.AuthType = AuthTypeDex
+				config.config.Cookie = testValue
+			},
+			wantErr: false,
+		},
+		{
+			name: "should return an error with roundTripper's AuthType set to AuthTypeDex empty auth fields",
+			fields: fields{
+				p: observatoriumRoundTripper{
+					config: config,
+					wrapped: observatoriumRoundTripper{
+						config:  config,
+						wrapped: pAPI.DefaultRoundTripper,
+					},
+				},
+			},
+			args: args{
+				request: req,
+			},
+			modifyFn: func(config *observatoriumRoundTripper) {
+				config.config.AuthType = AuthTypeDex
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			p := tt.fields.p
+			if tt.modifyFn != nil {
+				tt.modifyFn(&p)
+			}
+			_, err := p.RoundTrip(tt.args.request)
+			Expect(err != nil).To(Equal(tt.wantErr))
+		})
+	}
+}
+
+func Test_Query(t *testing.T) {
+	c := observabilityConfiguration
+	c.EnableMock = true
+	RegisterTestingT(t)
+
+	client, err := NewObservatoriumClient(c)
+	Expect(err).To(BeNil())
+	type fields struct {
+		c *Client
+	}
+	type args struct {
+		queryTemplate string
+		label         string
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		fields  fields
+		wantErr bool
+	}{
+		{
+			name: "should successfully execute Query",
+			fields: fields{
+				c: client,
+			},
+			args: args{
+				queryTemplate: "kafka_instance_connection_limit{%s}",
+				label:         "",
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			Expect(tt.fields.c.Query(tt.args.queryTemplate, tt.args.label).Err != nil).To(Equal(tt.wantErr))
```

Comment:
There are no test cases with `wantErr == true`

Reply:
I am not sure if its feasible to simulate `wantErr == true`

Reply:
I didn't look too much into the code, however if there are no cases were `wantErr` is `true` then I believe we should remove it and change the test to:
```suggestion
			Expect(tt.fields.c.Query(tt.args.queryTemplate, tt.args.label).Err).ToNot(HaveOccurred())
```

Reply:
thanks. I updated it to your most recent suggestion